src/nix/repl.cc

```diff
@@ -895,6 +895,22 @@ struct CmdRepl : InstallablesCommand
     CmdRepl(){
         evalSettings.pureEval = false;
     }
+    void prepare()
+    {
+        if (!settings.isExperimentalFeatureEnabled(Xp::Flakes) && !(file)) {
+            warn("future versions of Nix will require using `--file` to load a file");
+            if (this->_installables.size() > 1) {
+                warn("more than one input file is not currently supported");
+            }
```

Comment:
Afaik that’s just a limitation of the current CLI framework right?

(Not that we should change it now, but it’s good to keep in mind in case we need to revert this restriction)

Reply:
Yes, the current CLI takes "--file some-file" to change the meaning of all remaining installables to be attributes, thus we lost the ability to have multiple files. I doubt that is a high priority or common use case.